src/main/java/com/avispl/symphony/dal/avdevices/avcontrollers/planar/walldirector/WallDirectorCommunicator.java

```diff
@@ -0,0 +1,1020 @@
+        super();
+        this.setPort(this.getPort());
+        this.setCommandSuccessList(Collections.singletonList("\r\n"));
+        this.setCommandErrorList(Collections.singletonList("\r\n"));
+    }
+
+    /**
+     * {@inheritDoc}
+     * <p>
+     * Override the send() method to add a cool down delay time after every send command
+     */
+    @Override
+    public byte[] send(byte[] data) throws Exception {
+        try {
+            long currentTime = System.currentTimeMillis() - lastCommandTimestamp;
+            //check next command wait commandsCoolDownDelay time
+            if (currentTime < commandsCoolDownDelay) {
+                Thread.sleep(commandsCoolDownDelay - currentTime);
+            }
+            lastCommandTimestamp = System.currentTimeMillis();
+            if (logger.isDebugEnabled()) {
+                logger.debug(String.format("Issuing command %s, timestamp: %s", data, lastCommandTimestamp));
+            }
+            return super.send(data);
+        } finally {
+            logger.debug("send data command successfully");
+        }
+    }
+
+    @Override
+    public void controlProperty(ControllableProperty controllableProperty) throws Exception {
+
+    }
+
+    @Override
+    public void controlProperties(List<ControllableProperty> list) throws Exception {
+        if (CollectionUtils.isEmpty(list)) {
+            throw new IllegalArgumentException("ControllableProperties can not be null or empty");
+        }
+        for (ControllableProperty p : list) {
+            try {
+                controlProperty(p);
+            } catch (Exception e) {
+                logger.error(String.format("Error when control property %s", p.getProperty()), e);
+            }
+        }
+    }
+
+    @Override
+    public List<Statistics> getMultipleStatistics() throws Exception {
+        reentrantLock.lock();
+        try {
+            ExtendedStatistics extendedStatistics = new ExtendedStatistics();
+            Map<String, String> stats = new HashMap<>();
+            List<AdvancedControllableProperty> advancedControllableProperties = new ArrayList<>();
+            Map<String, String> dynamicStats = new HashMap<>();
+            if (countMonitoringAndControllingCommand == 0) {
+                countMonitoringAndControllingCommand = getNumberMonitoringAndControllingCommand();
+            }
+            if (!isEmergencyDelivery) {
+                convertConfigManagement();
+                failedMonitor.clear();
+                retrieveListID();
+                retrieveMonitoringAndControllingData();
+                if (failedMonitor.size() == countMonitoringAndControllingCommand) {
+                    throw new ResourceNotReachableException(String.format("There was an error while retrieving monitoring data for all %s properties.", countMonitoringAndControllingCommand));
+                }
+                destroyChannel();
+                populateGeneralData(stats, advancedControllableProperties);
```

Comment:
can you please also provide property filtering here? 
Check this for example 
https://github.com/AVISPL/dal-codecs-singlecodecs-polycom-videoos/blob/develop/src/main/java/com/avispl/dal/communicator/polycom/videoos/PolycomVideoOS.java

see property **displayPropertyGroups**